### PR TITLE
Add codebuild source credentials resources

### PR DESCRIPTION
**Pull request by dmytro-afanasiev**
Example
```yaml
policies:
  - name: codebuild-source-creds
    description: "Oauth not used for github/bitbucket"
    resource: aws.codebuild-source-credentials
    filters:
      - type: value
        key: serverType
        op: in
        value:
          - BITBUCKET
          - GITHUB
      - type: value
        key: authType
        op: ne
        value: OAUTH

```

### CODE REVIEW

File: c7n/resources/code.py [hunk 1]
```diff
@@ -142,6 +142,16 @@ class resource_type(TypeInfo):
     }
 
 
+@resources.register('codebuild-source-credentials')
```
Comment (kapilt) [inline c7n/resources/code.py:145]: thoughts on just saying credential or token (per arn identifier)

Comment (dmytro-afanasiev) [reply c7n/resources/code.py:145]: will "codebuild-credential" be appropriate?

Comment (dmytro-afanasiev) [reply c7n/resources/code.py:145]: Anyway we should indicate that this resource is related to codebuild service

Comment (kapilt) [reply c7n/resources/code.py]: credential works

File: c7n/resources/code.py [hunk 2]
```diff
@@ -142,6 +142,16 @@ class resource_type(TypeInfo):
     }
 
 
+@resources.register('credential')
```
Comment (kapilt) [inline c7n/resources/code.py:145]: codebuild prefix still needed to scope

**Comment by kapilt** [comment]
fwiw, in progress on the rename.

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**